### PR TITLE
Migration of Supervised contrative Learning tutorial to Keras3

### PR DESCRIPTION
**Pull request by LakshmiKalaKadali**
The dependency on tensorflow-addons (TFA) for npairs_loss has been removed. The Supervised Contrastive Loss is now implemented from scratch using keras.ops  
image

### CODE REVIEW

File: examples/vision/supervised-contrastive-learning.py [hunk 1]
```diff
@@ -22,22 +23,16 @@
...
 
-import tensorflow as tf
-import tensorflow_addons as tfa
-import numpy as np
-from tensorflow import keras
-from tensorflow.keras import layers
+import os
+
+# Set backend: "jax", "torch", or "tensorflow"
+os.environ["KERAS_BACKEND"] = "tensorflow"
```
Comment (hertschuh) [inline examples/vision/supervised-contrastive-learning.py]: Can you try with "jax"

Usually, it's a good way to check the conversion is fully done (for instance making sure the keras ops where used).

**Review by hertschuh — CHANGES_REQUESTED** [review]
Can you also reformat the code using `black`?

File: examples/vision/supervised-contrastive-learning.py (hunk 1 above)
Comment (LakshmiKalaKadali) [reply examples/vision/supervised-contrastive-learning.py]: Thank You. Sure will run with jax

File: examples/vision/supervised-contrastive-learning.py [hunk 2]
```diff
@@ -161,21 +156,38 @@ def create_classifier(encoder, trainable=True):
...
+
+        logits = ops.divide(
+            ops.matmul(feature_vectors, ops.transpose(feature_vectors)),
             self.temperature,
         )
-        return tfa.losses.npairs_loss(tf.squeeze(labels), logits)
+
+        # Create a mask to find positive pairs (images of same class)
+        labels = ops.cast(labels, "int32")
+        mask = ops.cast(ops.equal(labels, ops.transpose(labels)), "float32")
```
[the remark this answers is not in the document]
Comment (LakshmiKalaKadali) [reply examples/vision/supervised-contrastive-learning.py:175]: Valid

File: examples/vision/supervised-contrastive-learning.py (hunk 1 above)
Comment (LakshmiKalaKadali) [reply examples/vision/supervised-contrastive-learning.py]: Thank you for the review. I have run on Jax backend. It is running successfully.  Attached screenshot in the description. Reformatted as well. Thank You

File: examples/vision/supervised-contrastive-learning.py [hunk 3]
```diff
@@ -139,7 +135,7 @@ def create_classifier(encoder, trainable=True):
 classifier = create_classifier(encoder)
 classifier.summary()
 
-history = classifier.fit(x=x_train, y=y_train, batch_size=batch_size, epochs=num_epochs)
+# history = classifier.fit(x=x_train, y=y_train, batch_size=batch_size, epochs=num_epochs)
```
Comment (hertschuh) [inline examples/vision/supervised-contrastive-learning.py]: Why is this commented out?

Comment (LakshmiKalaKadali) [reply examples/vision/supervised-contrastive-learning.py]: As part of experimentation this line was commented earlier. Now uncommented. Thank You

**Review by hertschuh — APPROVED** [review]
Thanks!

Can you generate the ipynb and md files?

**Review by hertschuh — APPROVED** [review]
LGTM.

Can you regenerate the ipynb and md files?

**Comment by maitry63** [comment]
Hi @hertschuh , Thank you very much for taking the time to review this PR.

This work has been moved to PR #2398, where the files have been added correctly.
Closing this PR to keep the discussion consolidated in the new one.
Thank you again for your valuable review and feedback.
